src/Common/src/TypeSystem/Common/Utilities/GCPointerMap.cs

```diff
@@ -130,6 +130,25 @@ public override string ToString()
                 sb.Append(bit ? '1' : '0');
             return sb.ToString();
         }
+
+        public int CompareTo(object obj)
+        {
+            if (!(obj is GCPointerMap))
+                throw new ArgumentException();
+
+            GCPointerMap other = (GCPointerMap)obj;
+            if (_numCells != other._numCells)
+                return _numCells < other._numCells ? -1 : 1;
+
+            for (int i = 0; i < _gcFlags.Length; i++)
```

Comment:
Yes, we don't put garbage in the unused bits right now, so this works. Maybe we don't need to take that into consideration while sorting, but we might want to at least enforce unused bits are zero by asserting that.